server/superdesk/io/rfc822.py

```diff
@@ -23,11 +24,28 @@
 from superdesk.errors import IngestEmailError
 from bs4 import BeautifulSoup, Comment, Doctype
 import re
+from eve.utils import ParsedRequest
 
 
 logger = logging.getLogger(__name__)
 
 
+class UserNotRegisteredException(Exception):
+    pass
+
+
+email_regex = re.compile('^.*<(.*)>$')
+
+
+def get_user_by_email(field_from):
```

Comment:
or you are suggesting to split it to two routines, to parse e-mail field, and two get a user by email?